libbeat/processors/add_kubernetes_metadata/kubernetes.go

```diff
@@ -60,6 +73,11 @@ func New(cfg *common.Config) (processors.Processor, error) {
 		return nil, fmt.Errorf("fail to unpack the kubernetes configuration: %s", err)
 	}
 
+	// In cluster mode is already detected so ignore any given config
```

Comment:
Not sure about this.
If In cluster, but the user provided a kubeconfig, which would be very unexpected, imho we should obey what the user dictates.

Reply:
I think that we should ignore any given `kubeconfig` and just inform the user that `in_cluster` will be used. I will add a log.Info so as to make it clear to the user.